android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt

```diff
@@ -58,19 +58,29 @@ class RNGestureHandlerRootHelper(private val context: ReactContext, wrappedView:
   }
 
   internal inner class RootViewGestureHandler : GestureHandler<RootViewGestureHandler>() {
-    override fun onHandle(event: MotionEvent, sourceEvent: MotionEvent) {
+    private fun handleEvent(event: MotionEvent) {
       val currentState = state
+
       // we shouldn't stop intercepting events when there is an active handler already, which could happen when
       // adding a new pointer to the screen after a handler activates
       if (currentState == STATE_UNDETERMINED && (!shouldIntercept || orchestrator?.isAnyHandlerActive() != true)) {
         begin()
         shouldIntercept = false
       }
-      if (event.actionMasked == MotionEvent.ACTION_UP) {
+
+      if (event.actionMasked == MotionEvent.ACTION_UP || event.actionMasked == MotionEvent.ACTION_HOVER_EXIT) {
         end()
       }
     }
 
+    override fun onHandle(event: MotionEvent, sourceEvent: MotionEvent) {
+      handleEvent(event)
+    }
+
+    override fun onHandleHover(event: MotionEvent, sourceEvent: MotionEvent) {
+      handleEvent(event)
+    }
```

Comment:
```suggestion
    override fun onHandle(event: MotionEvent, sourceEvent: MotionEvent) = handleEvent(event)

    override fun onHandleHover(event: MotionEvent, sourceEvent: MotionEvent) = handleEvent(event)
```

Reply:
Done in [c1a9158](https://github.com/software-mansion/react-native-gesture-handler/pull/3462/commits/c1a91588266e9604bcc22ccd3f5b220383cf51a2)